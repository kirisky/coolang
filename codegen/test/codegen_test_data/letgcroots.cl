class A {
 c1:C;
 foo:Int;
 c2:C;
 
 setC1( newC1 : C ) : Object
  {
    {
      c1 <- newC1;
    }
  };
};

class B {
 foo:Int;
};

class C {
 a:A;
 setA( newA : A ) : Object
  {
    {
      a <- newA;
    }
  };
};

class Main inherits IO {
  main() : Object { {
    new A;
    new B;

    let
      a:A <- new A,
      b:B <- new B,
    in {
      let c:C <- new C in {
	    c.setA(a);
	    a.setC1(c);
	  };
    };
  } };
};
