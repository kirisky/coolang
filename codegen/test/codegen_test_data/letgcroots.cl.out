gc objs start
obj
  is_reachable=0
  typename=A
gc objs end

gc objs start
obj
  is_reachable=0
  typename=B
obj
  is_reachable=0
  typename=A
gc objs end

gc objs start
obj
  is_reachable=0
  typename=A
obj
  is_reachable=0
  typename=B
obj
  is_reachable=0
  typename=A
gc objs end

Inserting a root that points to:
obj
  is_reachable=0
  typename=A

Current GC roots:
Root that points to:
obj
  is_reachable=0
  typename=A
End of current GC roots

gc objs start
obj
  is_reachable=0
  typename=B
obj
  is_reachable=0
  typename=A
obj
  is_reachable=0
  typename=B
obj
  is_reachable=0
  typename=A
gc objs end

Inserting a root that points to:
obj
  is_reachable=0
  typename=B

Current GC roots:
Root that points to:
obj
  is_reachable=0
  typename=A
Root that points to:
obj
  is_reachable=0
  typename=B
End of current GC roots

gc objs start
obj
  is_reachable=0
  typename=C
obj
  is_reachable=0
  typename=B
obj
  is_reachable=0
  typename=A
obj
  is_reachable=0
  typename=B
obj
  is_reachable=0
  typename=A
gc objs end

Inserting a root that points to:
obj
  is_reachable=0
  typename=C

Current GC roots:
Root that points to:
obj
  is_reachable=0
  typename=A
Root that points to:
obj
  is_reachable=0
  typename=B
Root that points to:
obj
  is_reachable=0
  typename=C
End of current GC roots

Removing a root that points to:
obj
  is_reachable=0
  typename=C

Current GC roots:
Root that points to:
obj
  is_reachable=0
  typename=A
Root that points to:
obj
  is_reachable=0
  typename=B
End of current GC roots

Removing a root that points to:
obj
  is_reachable=0
  typename=B

Current GC roots:
Root that points to:
obj
  is_reachable=0
  typename=A
End of current GC roots

Removing a root that points to:
obj
  is_reachable=0
  typename=A

Current GC roots:
End of current GC roots

gc objs start
obj
  is_reachable=0
  typename=Object
obj
  is_reachable=0
  typename=C
obj
  is_reachable=0
  typename=B
obj
  is_reachable=0
  typename=A
obj
  is_reachable=0
  typename=B
obj
  is_reachable=0
  typename=A
gc objs end

